ros2_batch_job/__main__.py

```diff
@@ -380,6 +380,23 @@ def build_and_test(args, job):
     # xunit2 format is needed to make Jenkins xunit plugin 2.x happy
     with open('pytest.ini', 'w') as ini_file:
         ini_file.write('[pytest]\njunit_family=xunit2')
+    # check if packages have a pytest.ini file and add the xunit2
+    # format if it is not present
+    from pathlib import Path
+    from configparser import ConfigParser, NoOptionError
+    for path in Path('.').rglob('pytest.ini'):
+        config = ConfigParser()
+        config.read(str(path.resolve()))
+        try:
+            # only if xunit2 is set continue the loop with the file unpatched
+            if config.get('pytest', 'junit_family') == 'xunit2':
+                continue
+        except NoOptionError:
+            pass
+        print("xunit2 patch applied to " + str(path.resolve()))
```

Comment:
Nitpick: single quotes.

Again, why `.resolve()`? A relative path should be much more readable.

Reply:
5e1884c